// node-simconnect/dist/dto/bufferHelpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.readPBH = exports.readXYZ = exports.readLatLonAlt = exports.readWaypoint = exports.readMarkerState = exports.readInitPosition = void 0;
const InitPosition_1 = require("./InitPosition");
const MarkerState_1 = require("./MarkerState");
const Waypoint_1 = require("./Waypoint");
const LatLonAlt_1 = require("./LatLonAlt");
const XYZ_1 = require("./XYZ");
const PBH_1 = require("./PBH");
function readInitPosition(dataWrapper) {
    return readData(dataWrapper, new InitPosition_1.InitPosition());
}
exports.readInitPosition = readInitPosition;
function readMarkerState(dataWrapper) {
    return readData(dataWrapper, new MarkerState_1.MarkerState());
}
exports.readMarkerState = readMarkerState;
function readWaypoint(dataWrapper) {
    return readData(dataWrapper, new Waypoint_1.Waypoint());
}
exports.readWaypoint = readWaypoint;
function readLatLonAlt(dataWrapper) {
    return readData(dataWrapper, new LatLonAlt_1.LatLonAlt());
}
exports.readLatLonAlt = readLatLonAlt;
function readXYZ(dataWrapper) {
    return readData(dataWrapper, new XYZ_1.XYZ());
}
exports.readXYZ = readXYZ;
function readData(dataWrapper, obj) {
    obj.readFrom(dataWrapper);
    return obj;
}
function readPBH(dataWrapper) {
    return readData(dataWrapper, new PBH_1.PBH());
}
exports.readPBH = readPBH;
