// node-simconnect/dist/dto/InitPosition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Airspeed = exports.InitPosition = void 0;
var Airspeed;
(function (Airspeed) {
    Airspeed[Airspeed["Keep"] = -2] = "Keep";
    Airspeed[Airspeed["Cruise"] = -1] = "Cruise";
})(Airspeed || (Airspeed = {}));
exports.Airspeed = Airspeed;
class InitPosition {
    constructor() {
        this.latitude = 0;
        this.longitude = 0;
        this.altitude = 0;
        this.pitch = 0;
        this.bank = 0;
        this.heading = 0;
        this.onGround = false;
        this.airspeed = 0;
    }
    readFrom(buffer) {
        this.latitude = buffer.readFloat64();
        this.longitude = buffer.readFloat64();
        this.altitude = buffer.readFloat64();
        this.pitch = buffer.readFloat64();
        this.bank = buffer.readFloat64();
        this.heading = buffer.readFloat64();
        this.onGround = buffer.readInt32() !== 0;
        this.airspeed = buffer.readInt32();
    }
    writeTo(packetBuilder) {
        packetBuilder
            .putFloat64(this.latitude)
            .putFloat64(this.longitude)
            .putFloat64(this.altitude)
            .putFloat64(this.pitch)
            .putFloat64(this.bank)
            .putFloat64(this.heading)
            .putInt32(this.onGround ? 1 : 0)
            .putInt32(this.airspeed);
    }
}
exports.InitPosition = InitPosition;
