// node-simconnect/dist/enums/SimConnectException.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimConnectException = void 0;
var SimConnectException;
(function (SimConnectException) {
    SimConnectException[SimConnectException["NONE"] = 0] = "NONE";
    SimConnectException[SimConnectException["ERROR"] = 1] = "ERROR";
    SimConnectException[SimConnectException["SIZE_MISMATCH"] = 2] = "SIZE_MISMATCH";
    SimConnectException[SimConnectException["UNRECOGNIZED_ID"] = 3] = "UNRECOGNIZED_ID";
    SimConnectException[SimConnectException["UNOPENED"] = 4] = "UNOPENED";
    SimConnectException[SimConnectException["VERSION_MISMATCH"] = 5] = "VERSION_MISMATCH";
    SimConnectException[SimConnectException["TOO_MANY_GROUPS"] = 6] = "TOO_MANY_GROUPS";
    SimConnectException[SimConnectException["NAME_UNRECOGNIZED"] = 7] = "NAME_UNRECOGNIZED";
    SimConnectException[SimConnectException["TOO_MANY_EVENT_NAMES"] = 8] = "TOO_MANY_EVENT_NAMES";
    SimConnectException[SimConnectException["EVENT_ID_DUPLICATE"] = 9] = "EVENT_ID_DUPLICATE";
    SimConnectException[SimConnectException["TOO_MANY_MAPS"] = 10] = "TOO_MANY_MAPS";
    SimConnectException[SimConnectException["TOO_MANY_OBJECTS"] = 11] = "TOO_MANY_OBJECTS";
    SimConnectException[SimConnectException["TOO_MANY_REQUESTS"] = 12] = "TOO_MANY_REQUESTS";
    SimConnectException[SimConnectException["WEATHER_INVALID_PORT"] = 13] = "WEATHER_INVALID_PORT";
    SimConnectException[SimConnectException["WEATHER_INVALID_METAR"] = 14] = "WEATHER_INVALID_METAR";
    SimConnectException[SimConnectException["WEATHER_UNABLE_TO_GET_OBSERVATION"] = 15] = "WEATHER_UNABLE_TO_GET_OBSERVATION";
    SimConnectException[SimConnectException["WEATHER_UNABLE_TO_CREATE_STATION"] = 16] = "WEATHER_UNABLE_TO_CREATE_STATION";
    SimConnectException[SimConnectException["WEATHER_UNABLE_TO_REMOVE_STATION"] = 17] = "WEATHER_UNABLE_TO_REMOVE_STATION";
    SimConnectException[SimConnectException["INVALID_DATA_TYPE"] = 18] = "INVALID_DATA_TYPE";
    SimConnectException[SimConnectException["INVALID_DATA_SIZE"] = 19] = "INVALID_DATA_SIZE";
    SimConnectException[SimConnectException["DATA_ERROR"] = 20] = "DATA_ERROR";
    SimConnectException[SimConnectException["INVALID_ARRAY"] = 21] = "INVALID_ARRAY";
    SimConnectException[SimConnectException["CREATE_OBJECT_FAILED"] = 22] = "CREATE_OBJECT_FAILED";
    SimConnectException[SimConnectException["LOAD_FLIGHTPLAN_FAILED"] = 23] = "LOAD_FLIGHTPLAN_FAILED";
    SimConnectException[SimConnectException["OPERATION_INVALID_FOR_OJBECT_TYPE"] = 24] = "OPERATION_INVALID_FOR_OJBECT_TYPE";
    SimConnectException[SimConnectException["ILLEGAL_OPERATION"] = 25] = "ILLEGAL_OPERATION";
    SimConnectException[SimConnectException["ALREADY_SUBSCRIBED"] = 26] = "ALREADY_SUBSCRIBED";
    SimConnectException[SimConnectException["INVALID_ENUM"] = 27] = "INVALID_ENUM";
    SimConnectException[SimConnectException["DEFINITION_ERROR"] = 28] = "DEFINITION_ERROR";
    SimConnectException[SimConnectException["DUPLICATE_ID"] = 29] = "DUPLICATE_ID";
    SimConnectException[SimConnectException["DATUM_ID"] = 30] = "DATUM_ID";
    SimConnectException[SimConnectException["OUT_OF_BOUNDS"] = 31] = "OUT_OF_BOUNDS";
    SimConnectException[SimConnectException["ALREADY_CREATED"] = 32] = "ALREADY_CREATED";
    SimConnectException[SimConnectException["OBJECT_OUTSIDE_REALITY_BUBBLE"] = 33] = "OBJECT_OUTSIDE_REALITY_BUBBLE";
    SimConnectException[SimConnectException["OBJECT_CONTAINER"] = 34] = "OBJECT_CONTAINER";
    SimConnectException[SimConnectException["OBJECT_AI"] = 35] = "OBJECT_AI";
    SimConnectException[SimConnectException["OBJECT_ATC"] = 36] = "OBJECT_ATC";
    SimConnectException[SimConnectException["OBJECT_SCHEDULE"] = 37] = "OBJECT_SCHEDULE";
    SimConnectException[SimConnectException["JETWAY_DATA"] = 38] = "JETWAY_DATA";
    SimConnectException[SimConnectException["ACTION_NOT_FOUND"] = 39] = "ACTION_NOT_FOUND";
    SimConnectException[SimConnectException["NOT_AN_ACTION"] = 40] = "NOT_AN_ACTION";
    SimConnectException[SimConnectException["INCORRECT_ACTION_PARAMS"] = 41] = "INCORRECT_ACTION_PARAMS";
    SimConnectException[SimConnectException["GET_INPUT_EVENT_FAILED"] = 42] = "GET_INPUT_EVENT_FAILED";
    SimConnectException[SimConnectException["SET_INPUT_EVENT_FAILED"] = 43] = "SET_INPUT_EVENT_FAILED";
})(SimConnectException = exports.SimConnectException || (exports.SimConnectException = {}));
module.exports = {
    SimConnectException,
};
